packages/components/select/src/option-group.vue

```diff
@@ -57,25 +57,24 @@ export default defineComponent({
       children.value.some((option) => option.visible === true)
     )
 
+    const isOption = (node) =>
+      node.type?.name === 'ElOption' && node.component?.proxy
```

Comment:
According to the name semantics, the function should return a boolean value, perhaps the following adjustment will make it clearer.
```suggestion
      node.type?.name === 'ElOption' && !!node.component?.proxy
```